### DISCUSSION TITLE
UCAN for p2p Matrix

### DISCUSSION THREAD

**Discussion by matheus23** [ucan-wg/spec#83] [General]
The matrix folks want to go peer-to-peer: https://arewep2pyet.com/

They mention they need to authenticate matrix events with an "auth DAG":
![image](https://user-images.githubusercontent.com/1430958/179752883-fab32259-4678-4740-8a1d-66f095c57e5b.png)

Maybe UCAN could solve their needs?

It seems like they have a "stateful" kind of auth DAG, and may not be able to distinguish an "invocation" from a delegation. Might be interesting to look at their use cases more closely.

**Comment by bmann** [comment]
Thanks @matheus23 — I know @expede spoke with some Matrix folks, we should follow up with them on this.

**Comment by expede** [comment]
They're doing essentially a WebRTC mesh. Super cool, and should probably get included everywhere IMO. UCAN support would be good; and yeah I brought it up with Matrix when I chatted with them the week before last. Should probably follow up.
